packages/ui-components/src/Input/TextTypeInput.tsx

```diff
@@ -0,0 +1,176 @@
+import React from "react";
+import classNames from "classnames/bind";
+import { CommonInputProps, CommonInput } from "./CommonInput";
+import { isEmpty } from "lodash";
+import { FieldMetaState } from "react-final-form";
+import "./input.module.scss";
+
+export interface TextAndNumberProps
+  extends React.InputHTMLAttributes<HTMLInputElement> {
+  suffix?: JSX.Element;
+  onChange?: (
+    event: React.ChangeEvent<HTMLInputElement | HTMLTextAreaElement>,
+  ) => void;
+  onBlur?: (
+    event: React.FocusEvent<HTMLInputElement | HTMLTextAreaElement> | undefined,
+  ) => void;
+  onFocus?: (
+    event: React.FocusEvent<HTMLInputElement | HTMLTextAreaElement> | undefined,
+  ) => void;
+  // prop only used internally for implementation
+  type?: "text" | "email" | "password" | "number";
+  // this defines the input type rendered by this field
+  // for multiline text types, this will be <textarea>
+  // for text/email/password/number types this will be <input> with a type field
+  JSXInput?: JSX.Element;
+}
+
+export interface MultilineProps
+  extends React.DetailedHTMLProps<
+    React.TextareaHTMLAttributes<HTMLTextAreaElement>,
+    HTMLTextAreaElement
+  > {
+  suffix?: JSX.Element;
+}
+
+// the following props are used to implement {Email, Password, Number}Input
+// it is recommended to use those components instead of passing an "email" type to TextInput
+interface CustomProps extends React.InputHTMLAttributes<HTMLInputElement> {
+  prefixElement?: JSX.Element;
+  ariaLabel?: string;
+  ariaLabelledBy?: string;
+  autoFocus?: boolean;
+  tabIndex?: number;
+  onKeyDown?: (event: React.KeyboardEvent<HTMLInputElement>) => void;
+  onClick?: (event: React.MouseEvent<HTMLInputElement, MouseEvent>) => void;
+  forgotPasswordLinkElement?: JSX.Element;
+  // this prop is only used for internal implementation
+  // whether the password exists or not, should be indicated by New/Existing PasswordInput
+  existingPassword?: boolean;
+  //The actual type below as a parameter is called FieldValue
+  // not exported by react-final-form, but by google protobuf
+  // which would need to be added as a dependency
+  // since this prop comes from the Field component of react-final-form
+  // we can be fairly certain that the type won't be unexpected
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  meta?: FieldMetaState<any>;
```

Comment:
I had typed it as `any` because I wasn't sure what `FieldValue` was in `FieldRenderProps` (which passed that value to `FieldMetaState`) - I looked into it and it's defined by the HTML Input component we pass in. Updated the code to reflect this.